services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -130,11 +130,17 @@ async def list_files_in_dataset(
         return data
 
     async def list_files(  # noqa C901
-        self, user_id: UserID, *, expand_dirs: bool, uuid_filter: str = ""
+        self,
+        user_id: UserID,
+        *,
+        expand_dirs: bool,
+        uuid_filter: str = "",
+        project_id: ProjectID | None = None,
     ) -> list[FileMetaData]:
         """
         expand_dirs `False`: returns one metadata entry for each directory
         expand_dirs `True`: returns all files in each directory (no directories will be included)
+        project_id: If passed, only list files associated with that project_id
```

Comment:
what happens if I pass a uuid_filter and a project_id?

Reply:
Then only files whose `object_name` match the `uuid_filter` (in the sense of postgres ilike) and which are part of the project with id `project_id` will be listed. I have added some documentation here which should hopefully make this more clear.